electrum/plugin.py

```diff
@@ -72,22 +76,45 @@ def __init__(self, config: SimpleConfig, gui_name):
         self.plugins = {}  # type: Dict[str, BasePlugin]
         self.internal_plugin_metadata = {}
         self.external_plugin_metadata = {}
+        self.installed_plugin_metadata = {}
         self.gui_name = gui_name
         self.device_manager = DeviceMgr(config)
         self.user_pkgpath = os.path.join(self.config.electrum_path_root(), 'plugins')
         if not os.path.exists(self.user_pkgpath):
             os.mkdir(self.user_pkgpath)
+        self.init_trusted_plugins()
         self.find_internal_plugins()
         self.find_external_plugins()
+        self.find_installed_plugins()
         self.load_plugins()
         self.add_jobs(self.device_manager.thread_jobs())
         self.start()
 
     @property
     def descriptions(self):
-        return dict(list(self.internal_plugin_metadata.items()) + list(self.external_plugin_metadata.items()))
+        # if a plugin was installed but is no longer in external_plugin_metadata,
+        # it will be returned by this method
+        d = {}
+        d.update(self.internal_plugin_metadata)
+        d.update(self.external_plugin_metadata)
+        d.update(self.installed_plugin_metadata)
+        return d
+
+    def is_installed(self, name):
+        return name in self.installed_plugin_metadata
+
+    def can_be_upgraded(self, name):
+        from packaging.version import Version
```

Comment:
`packaging` is a 3rd-party library, not part of the stdlib.
Looks like we indirectly depend on it currently via qdarkstyle, but if possible I think it would still be preferable to avoid making it a direct dependency also. Is it needed here?
```
  - qdarkstyle [required: >=3.2, installed: 3.2.3]
    - qtpy [required: >=2, installed: 2.1.0]
      - packaging [required: Any, installed: 23.1]
```

Reply:
For trivial int-only version numbers (are those not sufficient here?) we could use `electrum.util.versiontuple`.
For more advanced strings there is `electrum._vendor.distutils.version.StrictVersion` (though I don't know how exactly it compares to `packaging.version.Version`).

Reply:
fixed

Reply:
1000

Reply:
hamedmensvri@gmail.com